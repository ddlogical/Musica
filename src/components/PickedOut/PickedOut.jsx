import "./PickedOut.scss";
import { Link } from "react-router-dom";
import { shallowEqual, useSelector } from "react-redux";
import Icon from '../Icon';
import { memo } from 'react';

function PickedOut() {
    
    const pickedOutLength = useSelector(state => state.store.pickedOut.length, shallowEqual);

    return (
        <Link to='/picked-out' className='picked-out'>

            <Icon color="#fff" type="star-fill" />
            <span>({pickedOutLength})</span>
        </Link>
    );
}


export default memo(PickedOut);