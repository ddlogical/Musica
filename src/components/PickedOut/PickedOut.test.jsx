import { MemoryRouter } from "react-router-dom";
import { renderWithProviders } from "../../test.utils";
import '@testing-library/jest-dom';

import PickedOut from './';

test("PickedOut shapshot", () => {
    const view = renderWithProviders(<MemoryRouter><PickedOut /></MemoryRouter>, {preloadedState: {
        store: {}
      }});
      expect(view).toMatchSnapshot();
})