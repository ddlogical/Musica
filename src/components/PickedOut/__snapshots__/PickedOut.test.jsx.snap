// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickedOut shapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="picked-out"
        href="/picked-out"
      >
        <svg
          fill="#fff"
          height="24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m7.825 19.225 1.125-4.75-3.7-3.175 4.85-.425L12 6.4l1.925 4.475 4.825.425-3.675 3.175 1.125 4.75L12 16.7Z"
          />
        </svg>
        <span>
          (
          0
          )
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="picked-out"
      href="/picked-out"
    >
      <svg
        fill="#fff"
        height="24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m7.825 19.225 1.125-4.75-3.7-3.175 4.85-.425L12 6.4l1.925 4.475 4.825.425-3.675 3.175 1.125 4.75L12 16.7Z"
        />
      </svg>
      <span>
        (
        0
        )
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
